src/Async/index.js

```diff
@@ -323,6 +323,42 @@ function Async(fn) {
     }
   }
 
+  function applyTo(method) {
+    return function (liftedApply) {
+      if (!isSameType(Async, liftedApply)) {
+        throw new TypeError(`Async.${method}: Async required`)
+      }
+
+      // liftedApply is Async[A => B]
+      return Async(function (reject, resolve) {
+        let resolvedA = null
+        let resolvedF = null
+        let cancelled = false
+
+        const rejectOnce = once(reject)
+
+        // todo: not sure if this is necessary
+        const cancel = () => { cancelled = true }
+
+        function resolveBoth() {
+          if (!cancelled && resolvedA && resolvedF) {
+            resolve(resolvedF(resolvedA))
```

Comment:
Like this better then the `compose(resolve, resolvedF)(resolvedA)` that was implemented in the original `ap`. This is better for the stack on these lazy types.

Will also change these silly compositions in `ap` when we address the "pointless" cancelled flag.